src/components/PaginationButtons.js

```diff
@@ -0,0 +1,186 @@
+const React = require('react');
+
+const ButtonGroup = require('./ButtonGroup');
+
+const StyleConstants = require('../constants/Style');
+
+const { buttonTypes } = require('../constants/App');
+
+const PaginationButtons = React.createClass({
+  propTypes: {
+    currentPage: React.PropTypes.number.isRequired,
+    onClick: React.PropTypes.func,
+    pageRange: React.PropTypes.number,
+    primaryColor: React.PropTypes.string,
+    style: React.PropTypes.object,
+    totalPages: React.PropTypes.number.isRequired,
+    type: React.PropTypes.oneOf(buttonTypes)
+  },
+
+  getDefaultProps () {
+    return {
+      currentPage: 1,
+      pageRange: 9,
+      primaryColor: StyleConstants.Colors.PRIMARY,
+      totalPages: 1,
+      type: 'primaryOutline'
+    };
+  },
+
+  _handleButtonClick (buttonClicked) {
+    const { currentPage, totalPages } = this.props;
+    let goToPage = buttonClicked;
+
+    if (buttonClicked === 'prev') {
+      goToPage = currentPage > 1 ? currentPage - 1 : 1;
+    } else if (buttonClicked === 'next') {
+      goToPage = currentPage < totalPages ? currentPage + 1 : totalPages;
+    }
+
+    this.props.onClick(goToPage);
+  },
+
+  _getPrevButton () {
+    const disabledStyle = this.props.currentPage <= 1 && this.styles().disabled;
+    const style = Object.assign({}, this.styles().component, disabledStyle);
+
+    return {
+      icon: 'caret-left',
+      onClick: this._handleButtonClick.bind(null, 'prev'),
+      style
+    };
+  },
+
+  _getNextButton () {
+    const disabledStyle = this.props.currentPage >= this.props.totalPages && this.styles().disabled;
+    const style = Object.assign({}, this.styles().component, disabledStyle);
+
+    return {
+      icon: 'caret-right',
+      onClick: this._handleButtonClick.bind(null, 'next'),
+      style
+    };
+  },
+
+  _getStartingPage (currentPage, maxStartingPage, staticSet, startingSet, middleSet) {
+    let startingPage = 1;
+
+    if (!staticSet && !startingSet) {
+      if (middleSet) {
+        startingPage = currentPage;
+      } else {
+        startingPage = maxStartingPage;
+      }
+    }
+
+    return startingPage;
+  },
+
+  _getEndingPage (totalPages, pageRange, startingSet, staticSet, middleSet, startingPage) {
+    let endingPage = totalPages;
+
+    if (!staticSet) {
+      if (startingSet) {
+        endingPage = pageRange - 2;
+      } else if (middleSet) {
+        endingPage = startingPage + pageRange - 5;
+      }
+    }
+
+    return endingPage;
+  },
+
+  _addFirstPageButtons () {
+    const style = this.styles();
+
+    return [{
+      onClick: this._handleButtonClick.bind(null, 1),
+      style: style.component,
+      text: '1'
+    }, {
+      icon: 'kabob_horizontal',
+      style: Object.assign({}, style.component, style.disabled)
```

Comment:
Changing the button type would mess with the CSS, including the borders. We want the button group to have a single type to avoid CSS weirdness but maybe that's something we could add in the future?